.github/workflows/python_check.yml

```diff
@@ -0,0 +1,79 @@
+name: Python Check
+
+on:
+  push:
+    branches: [ "main" ]
+    paths:
+      - '.github/workflows/python_check.yml'
+      - 'bindings/python/src/**'
+      - 'bindings/python/tests/**'
+      - 'bindings/python/pyproject.toml'
+      - 'bindings/python/pdm.lock'
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - '.github/workflows/python_check.yml'
+      - 'bindings/python/src/**'
+      - 'bindings/python/tests/**'
+      - 'bindings/python/pyproject.toml'
+      - 'bindings/python/pdm.lock'
+  merge_group:
+  workflow_dispatch:
+
+jobs:
+  lint:
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```

Comment:
Python lint might not need to run with different versions. (Though they do have some differences)